src/integrations/ga/SplitToGa.js

```diff
@@ -1,10 +1,108 @@
-// @TODO implement real module
-// eslint-disable-next-line no-unused-vars
-function SplitToGaFactory(options) {
-  return {
-    // eslint-disable-next-line no-unused-vars
-    queue: function (data, type) {/* noop for now */ }
-  };
+import logFactory from '../../utils/logger';
+import { uniq } from '../../utils/lang';
+import { SPLIT_IMPRESSION, SPLIT_EVENT } from '../../utils/constants';
+const log = logFactory('splitio-split-to-ga');
+
+class SplitToGa {
+
+  static defaultFilter() { return true; }
+
+  static defaultMapper({ type, payload }) {
+    switch (type) {
+      case SPLIT_IMPRESSION:
+        return {
+          hitType: 'event',
+          eventCategory: 'split-impression',
+          eventAction: payload.impression.feature,
+          eventLabel: payload.impression.treatment,
+          nonInteraction: true,
+        };
+      case SPLIT_EVENT:
+        return {
+          hitType: 'event',
+          eventCategory: 'split-event',
+          eventAction: payload.eventTypeId,
+          eventValue: payload.value,
+          nonInteraction: true,
+        };
+    }
+    return null;
+  }
+
+  static getGa() {
+    return typeof window !== 'undefined' ? window[window['GoogleAnalyticsObject'] || 'ga'] : undefined;
+  }
+
+  /**
+   * Validates if a given object is a UniversalAnalytics.FieldsObject instance, and logs a warning if not.
+   * It checks that the object contains a `hitType`, since it is the minimal field required to send the hit
+   * and avoid the GA error `No hit type specified. Aborting hit.`.
+   * Other validations (e.g., an `event` hitType must have a `eventCategory` and `eventAction`) are handled
+   * and logged (as warnings or errors depending the case) by GA debugger, but the hit is sent anyway.
+   *
+   * @param {UniversalAnalytics.FieldsObject} fieldsObject object to validate.
+   * @returns {boolean} Whether the data instance is a valid FieldsObject or not.
+   */
+  static validateFieldsObject(fieldsObject) {
+    if (fieldsObject && fieldsObject.hitType)
+      return true;
+
+    log.warn('your custom mapper returned an invalid FieldsObject instance. It must be an object with at least a `hitType` field.');
+    return false;
+  }
+
+  constructor(options) {
+
+    // Check if `ga` object is available
+    if (typeof SplitToGa.getGa() !== 'function') {
+      // @TODO review the following warning message
+      log.warn('`ga` command queue not found. No hits will be sent.');
+      // Return an empty object to avoid creating a SplitToGa instance
+      return {};
+    }
+
+    this.filter = options && typeof (options.filter) === 'function' ?
```

Comment:
From this line going forward, all the statements on the constructor depend on the "options" variable being truthy. Use a top level if statement to check that, then in it you only check for the property of the options you care on each assignment.